ton_client/src/debot/dengine.rs

```diff
@@ -179,6 +182,51 @@ impl DEngine {
         }
     }
 
+    pub async fn send(&mut self, source: String, func_id: u32, params: String) -> ClientResult<()> {
+        debug!("send from {} id = {} params = {}", source, func_id, params);
+        let params = serde_json::from_str(&params)
+            .map_err(|e| Error::invalid_json_params(e) )?;
+        let abi = Contract::load(self.raw_abi.as_bytes()).unwrap();
+        let func_name = &abi.function_by_id(func_id, true)
+            .map_err(|e| Error::invalid_function_id(e) )?
+            .name;
+        
+        let msg_params = ParamsOfEncodeMessageBody {
+            abi: self.abi.clone(),
+            signer: Signer::None,
+            processing_try_index: None,
+            is_internal: true,
+            call_set: CallSet::some_with_function_and_input(func_name, params).unwrap(),
+        };
+        let body = encode_message_body(self.ton.clone(), msg_params).await?.body;
+
+        let src_addr = account_decode(&source)?;
+        let dst_addr = account_decode(&self.addr)?;
+        let mut msg = Message::with_int_header(
+            InternalMessageHeader::with_addresses(src_addr, dst_addr, Default::default())
+        );
+        let (_, body_cell) = deserialize_cell_from_base64(&body, "message body")?;
+        msg.set_body(body_cell.into());
+        let msg_base64 = serialize_object_to_base64(&msg, "message")?;
+        
+        let run_result = run_tvm(
+            self.ton.clone(),
+            ParamsOfRunTvm {
+                account: std::mem::take(&mut self.state),
+                message: msg_base64,
+                abi: Some(self.abi.clone()),
+                execution_options: None,
+            },
+        ).await?;
+        let mut run_output = RunOutput::new(
+            run_result.account,
+            run_result.decoded.unwrap().output,
```

Comment:
Is `decoded` field guaranteed to have proper value? Better check to be sure

Reply:
Fixed